.github/workflows/workflow.yml

```diff
@@ -1,8 +1,11 @@
 name: Main workflow
 
 on:
-  - pull_request
-  - push
+  pull_request:
+  push:
+  # Prime the caches every Monday
```

Comment:
Obviously, a user could choose a different schedule, and update the date-based cache key generation accordingly. The point is to use the cron job to ensure that the default branch's cache is primed in advance of the first push or PR.